src/EventStore.Core/Services/Transport/Enumerators/Enumerator.StreamSubscription.cs

```diff
@@ -171,22 +273,29 @@ public class StreamSubscription<TStreamId> : StreamSubscription {
 							}
 
 							if (completed.IsEndOfStream) {
-								GoLive(StreamRevision.FromInt64(completed.NextEventNumber));
-								return;
+								if (completed.Events.Length > 0) {
+									catchupCompletionTcs.TrySetResult(completed.Events[^1].OriginalEventNumber);
+									return;
+								}
+
+								// the read returned an empty result - we don't have the last event number.
+								// we could derive it from `completed.NextEventNumber` but it's safer to stay in catch-up
+								// and retry the read as it's an edge case that shouldn't happen often.
+								await Task.Delay(TimeSpan.FromMilliseconds(500), ct);
```

Comment:
I'm not sure if it's a problem necessarily, but if you have a stream that isn't deleted but has no events (for example, with `$maxAge` or `$tb`) then the reads will continue to be retried until a new event is written (if ever)

Reply:
hey, thanks, i hadn't thought about these cases. There is an additional case where this can happen:
- if you subscribe to a checkpoint that's ahead of the last event number

I'll try to adjust things so that we can switch directly to live in these cases.

Reply:
fixed